specification/imagebuilder/resource-manager/Microsoft.VirtualMachineImages/stable/2021-10-01/imagebuilder.json

```diff
@@ -0,0 +1,1553 @@
+          }
+        },
+        "excludeFromLatest": {
+          "type": "boolean",
+          "default": false,
+          "description": "Flag that indicates whether created image version should be excluded from latest. Omit to use the default (false)."
+        },
+        "storageAccountType": {
+          "type": "string",
+          "enum": [
+            "Standard_LRS",
+            "Standard_ZRS"
+          ],
+          "x-ms-enum": {
+            "name": "SharedImageStorageAccountType",
+            "modelAsString": true
+          },
+          "description": "Storage account type to be used to store the shared image. Omit to use the default (Standard_LRS)."
+        }
+      },
+      "required": [
+        "galleryImageId",
+        "replicationRegions"
+      ]
+    },
+    "ImageTemplateVhdDistributor": {
+      "x-ms-discriminator-value": "VHD",
+      "description": "Distribute via VHD in a storage account.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ImageTemplateDistributor"
+        }
+      ]
+    },
+    "PlatformImagePurchasePlan": {
+      "type": "object",
+      "description": "Purchase plan configuration for platform image.",
+      "properties": {
+        "planName": {
+          "type": "string",
+          "description": "Name of the purchase plan."
+        },
+        "planProduct": {
+          "type": "string",
+          "description": "Product of the purchase plan."
+        },
+        "planPublisher": {
+          "type": "string",
+          "description": "Publisher of the purchase plan."
+        }
+      },
+      "required": [
+        "planName",
+        "planProduct",
+        "planPublisher"
+      ]
+    },
+    "VirtualNetworkConfig": {
+      "type": "object",
+      "description": "Virtual Network configuration.",
+      "properties": {
+        "subnetId": {
+          "type": "string",
+          "description": "Resource id of a pre-existing subnet."
+        },
+        "proxyVmSize": {
+          "type": "string",
+          "default": "",
+          "description": "Size of the proxy virtual machine used to pass traffic to the build virtual machine. Omit or specify empty string to use the default (Standard_A1_v2)."
+        }
+      }
+    },
+    "ImageTemplateVmProfile": {
+      "type": "object",
+      "description": "Describes the virtual machine used to build, customize and capture images",
+      "properties": {
+        "vmSize": {
+          "type": "string",
+          "default": "",
+          "description": "Size of the virtual machine used to build, customize and capture images. Omit or specify empty string to use the default (Standard_D2ds_v4)."
+        },
+        "osDiskSizeGB": {
+          "type": "integer",
+          "minimum": 0,
+          "default": 0,
+          "description": "Size of the OS disk in GB. Omit or specify 0 to use Azure's default OS disk size."
+        },
+        "userAssignedIdentities": {
```

Comment:
>userAssignedIdentities [](http://example.com/codeflow?start=9&length=22)

User managed identities follow a specific format, any reason you are not following this: [https://armwiki.azurewebsites.net/authorization/managed_identities/MSIOnboardingUserAssigned.html#update-resource-definition](https://armwiki.azurewebsites.net/authorization/managed_identities/MSIOnboardingUserAssigned.html#update-resource-definition)

Reply:
@sumit-kalra can you take a look at this?

Reply:
These are two different sets of identities.
- The ones on line 1164 are the typical identities that will be associated with this ARM resource (Microsoft.VirtualMachineImages.ImageTemplates). The RP may take actions on behalf of the customer using these identities.
- The ones mentioned here (line 986) are identities that are *not* used directly by this RP and are also *never* even interpreted by this RP. IOW, this RP never gets tokens for these identities and never tries to impersonate these identities. These are simply passed on as strings to an ARM template that deploys a VM.

Reply:
@jorgecotillo does this answer your question?